fnet/src/tests/locking/lockspeed.cpp

```diff
@@ -3,14 +3,30 @@
 #include <vespa/fnet/fnet.h>
 #include "dummy.h"
 
+class SpinLock {
+private:
+    std::atomic_flag _lock;
+public:
+    SpinLock() : _lock(false) {}
+    void lock() {
+        while (_lock.test_and_set(std::memory_order_acquire)) {
+            // spin
```

Comment:
Would be interesting to have a (perhaps template `constexpr if`'d) variant of this that invokes the [_mm_pause](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=4067,602,2978,4071,3884&techs=SSE,SSE2,SSE3,SSSE3,SSE4_1,SSE4_2,AVX,AVX2,Other&text=pause) intrinsic inside the spin loop. Avoids CPU spinning hot in case of contention.

Reply:
This spinlock is just a simple prototype to measure the minimal time it takes to obtain an uncontended lock as seen from the locking thread. Adding something to the body of the while should have minimal impact on performance since it is never run, but might produce a slightly higher lower bound on the actual overhead of the lock. It would make sense to benchmark both nice and naughty spinlocks if we add an actual implementation to vespalib.

The unknown dimension here is how busy looping interacts with hyper-threading and how additional atomic operations interact with other CPU cores. Hence the measured performance gain is very much an upper bound.